setup.py

```diff
@@ -31,13 +31,19 @@
     "uncertainties>=3.1.4",
     "optics-functions>=0.1.0",
     "turn_by_turn>=0.4.0",
+    "requests>=2.27.0",
```

Comment:
I don't think it's such a big requirement. and otherwise you would also need to mock the import to assure no crashes on importing modules etc. I feel it's saver there.